src/elements/SingleReview.jsx

```diff
@@ -1,38 +1,49 @@
 import { useState, useEffect } from "react";
-import { patchReviewVotesById, getCommentsById } from "../utils/api";
+import {
+  patchReviewVotesById,
+  getCommentsById,
+  postCommentById,
+  getReviewById,
+} from "../utils/api";
 import { CommentCard } from "./CommentCard";
 import moment from "moment/moment";
+import { useParams } from "react-router-dom";
 
-export const SingleReview = ({
-  reviewData: {
-    review_id,
-    title,
-    owner,
-    created_at,
-    category,
-    designer,
-    review_img_url,
-    review_body,
-    votes,
-    comment_count,
-  },
-}) => {
-  const [localVotes, setLocalVotes] = useState(votes);
+export const SingleReview = ({ username }) => {
+  const { id } = useParams();
+
+  const [review, setReview] = useState();
+  const [reviewLoaded, setReviewLoaded] = useState(false);
+
+  const [localVotes, setLocalVotes] = useState(0);
   const [processingVote, setProcessingVote] = useState(false);
+
   const [comments, setComments] = useState([]);
   const [commentsLoaded, setCommentsLoaded] = useState(false);
+  const [postingComment, setPostingComment] = useState(false);
+  const [newCommentText, setNewCommentText] = useState("");
 
   useEffect(() => {
```

Comment:
If you're using a loading pattern I'd reset the loading patterns to false at the beginning of the useEffect because whenever we're triggering this useEffect we're reloading data